src/model.ts

```diff
@@ -157,27 +162,45 @@ export class JSORMBase {
   private static _credentialStorage?: CredentialStorage
 
   static get credentialStorage(): CredentialStorage {
-    if (!this._credentialStorage) {
-      if (!this._credentialStorageBackend) {
-        if (this.jwtStorage && typeof localStorage !== "undefined") {
-          this._credentialStorageBackend = localStorage
-        } else {
-          this._credentialStorageBackend = new NullStorageBackend()
-        }
-      }
-
-      this._credentialStorage = new CredentialStorage(
-        this.jwtStorage,
-        this._credentialStorageBackend
-      )
+    if (this.baseClass === undefined) {
+      throw new Error(`No base class for ${this.name}`)
     }
 
-    return this._credentialStorage
+    if (!this._credentialStorage) {
+      return this._initializeCredentialStorage()
+    } else {
+      return this._credentialStorage
+    }
   }
 
   static set credentialStorageBackend(backend: StorageBackend | undefined) {
+    if (this.baseClass === JSORMBase) {
+      throw new Error(`Cannot access credentials on JSORMBase`)
+    }
+
     this._credentialStorageBackend = backend
-    this._credentialStorage = undefined
+    this._initializeCredentialStorage()
+  }
+
+  private static _initializeCredentialStorage() {
+    if (this.baseClass === JSORMBase) {
+      throw new Error(`Cannot access credentials on JSORMBase`)
+    }
+
+    if (!this._credentialStorageBackend) {
+      if (this.jwtStorage && typeof localStorage !== "undefined") {
```

Comment:
Just noting that `this.jwtStorage` could be `false` and we'd still create an in-memory backend. I don't care too much but heads up.

Reply:
yeah, that's the intended functionality. Essentially In-memory store is the same as not storing it.